mbuild/formats/lammpsdata.py

```diff
@@ -524,3 +488,121 @@ def write_lammpsdata(structure, filename, atom_style='full',
                     data.write('{:d}\t{:d}\t{:d}\t{:d}\t{:d}\t{:d}\n'.format(
                         i+1,improper_types[i],improper[2],
                         improper[1],improper[0],improper[3]))
+
+def _get_bond_types(structure, bonds, sigma_conversion_factor, 
+        epsilon_conversion_factor):
+    unique_bond_types = dict(enumerate(set([(round(bond.type.k*(
+        sigma_conversion_factor**2/epsilon_conversion_factor),3),
+                                             round(bond.type.req/sigma_conversion_factor,3),
+                                             tuple(sorted((bond.atom1.type,bond.atom2.type)))
+                                             ) for bond in structure.bonds])))
+    unique_bond_types = OrderedDict([(y,x+1) for x,y in unique_bond_types.items()])
+    bond_types = [unique_bond_types[(round(bond.type.k*
+        (sigma_conversion_factor**2/epsilon_conversion_factor),3),
+                                     round(bond.type.req/sigma_conversion_factor,3),
+                                     tuple(sorted((bond.atom1.type,bond.atom2.type)))
+                                     )] for bond in structure.bonds]
+    return bond_types, unique_bond_types
+
+def _get_angle_types(structure, use_urey_bradleys,
+        sigma_conversion_factor, epsilon_conversion_factor):
+    if use_urey_bradleys:
+        charmm_angle_types = []
+        for angle in structure.angles:
+            ub_k = 0
+            ub_req = 0
+            for ub in structure.urey_bradleys:
+                if (angle.atom1, angle.atom3) == (ub.atom1, ub.atom2):
+                    ub_k = ub.type.k
+                    ub_req = ub.type.req
+            charmm_angle_types.append((round(angle.type.k*
+                (sigma_conversion_factor/epsilon_conversion_factor),3), 
```

Comment:
For urey bradleys, I think only the K_ub needs the sigma conversion factor. The angle constant doesn't have distance units i think